tdms/src/interpolate_Hfield.cpp

```diff
@@ -1,161 +1,235 @@
- *
- * Interpolating the magnetic field requires us to first interpolate to obtain points parallel to the centre of the Yee cell, then again in the direction corresponding to the field component we are interested in.
- * This function determines which of the remaining two dimensions, labelled d0 and d1, is optimal for the first set of interpolations.
- *
- * (d0,d1) represents any combination of (x,y), (x,z), or (z,y) dimensions.
- *
- * @param n_cells_d0,n_cells_d1 The maxmium number of Yee cells in dimensions d0 and d1
- * @param cid0,cid1 The component of the Yee cell index in dimensions d0 and d1
- * @return true Dimension d0 should be used.
- * @return false Dimension d1 should be used. In the event of a tie, prefer this dimension.
- */
-bool determine_second_dim(int n_cells_d0, int n_cells_d1, int cid0, int cid1) {
-    // if we are doing a 2D simulation, one of the dimensions might be zero
-    // in which case, return the other
-    if (n_cells_d0 == 0) return false;
-    else if (n_cells_d1 == 0) return true;
-    else {
-        // so we have a 3D simulation
-        interp_scheme d0_scheme = determineInterpScheme(n_cells_d0, cid0);
-        interp_scheme d1_scheme = determineInterpScheme(n_cells_d1, cid1);
-        // we now need to determine which scheme is the better scheme
-        return better_scheme(d0_scheme, d1_scheme);
-    }
-}
+    The case when y and z have their roles reversed is similar.
+    */
 
-/**
- * @brief Interpolate the x-component of the magnetic field to the centre of Yee cell i,j,k
- *
- * Interpolation for the magnetic field must be done across two dimensions - first we must interpolate to obtain nearby field values parallel to the centre of the Yee cell, then we must interpolate again in another dimension to the centre of the Yee cell itself.
- *
- * @param[in] Hxy,Hxz Split components of the Yee cell
- * @param[in] i,j,k Yee cell index
- * @param[in] I,J,K Number of Yee cells in the i,j,k directions respectively
- * @param[out] Hx Interpolated value of the Hx field at centre of Yee cell i,j,k
- */
-void interpolateTimeDomainHx(double ***Hxy, double ***Hxz, int i, int j, int k, int I, int J, int K, double *Hx) {
-    // which interpolation scheme do we want to use in the x-direction?
-    interp_scheme x_scheme = determineInterpScheme(I, i);
+    // determine the z-direction scheme
+    const interpScheme &z_scheme = best_interp_scheme(K, k);
+    // determine the y-direction scheme
+    const interpScheme &y_scheme = best_interp_scheme(J, j);
 
-    // this interpolation scheme will determine how many times we need to interpolate in the other dimension, and the range of x-Yee cells to consider
-    int i_low, i_high, d_low, d_high;
-    // we will also need to store the result of the interpolation in the first dimension
-    double *first_interp_vals;
-    double *other_dim_cell_data;
+    // this data will be passed to the second interpolation scheme
+    double pass_to_second_scheme[8];
```

Comment:
maybe a slightly more consistent variable name might be good?  i.e. `data_for_second_scheme` rather than `pass_to_second_scheme`